src/shaders/workingRef-vert.glsl

```diff
@@ -0,0 +1,238 @@
+// of surrounding locations so it's a way to keep consistent which noise function is being used as 'the' noise 
+// function outisde of the 3D multi-octave lattice-value noise function
+float currNoiseFunc(float in1, float in2, float in3) {
+    return noiseFunc_2(in1, in2, in3);
+}
+
+// LINEAR INTERP FUNCTION for interpolating between two different values with a given t linearly defined % to get 
+// destination location between
+float linearInterp(float in1, float in2, float t) {
+	return (in1 * (1.0 - t + in2 * t));
+}
+    
+// COS INTERP FUNCTION for interpolating between two different values with a given t linearly defined % to get 
+// destination location between
+// uses the linear interp as part of it
+float cosInterpFunc(float in1, float in2, float t) {
+	float PI = 3.14159265;
+    float cos_t = (1.0 - cos(t * PI)) * 0.5;
+    return linearInterp(in1, in2, cos_t);
+}
+    
+// PRODUCES NOISE VALUE FOR POINT by interpolating surrounding lattice values (8 corner points)
+float interpNoiseByCorners(float x, float y, float z) {
+    float interval = 13.0;  
+
+    float xDist = x - mod(x, interval);
+    float yDist = y - mod(y, interval);
+    float zDist = z - mod(z, interval);
+
+    float xLow = x - xDist;
+    float yLow = y - yDist;
+    float zLow = z - zDist;
+
+    float xHigh = xLow + interval;
+    float yHigh = yLow + interval;
+    float zHigh = zLow + interval;
+
+    //upper cube check loc
+    float tl_up = currNoiseFunc(xLow,  yHigh, zHigh);
+    float tr_up = currNoiseFunc(xHigh, yHigh, zHigh);
+    float bl_up = currNoiseFunc(xLow,  yLow,  zHigh);
+    float br_up = currNoiseFunc(xHigh, yLow,  zHigh);
+    //lower cube check loc
+    float tl_bot = currNoiseFunc(xLow,  yHigh, zLow);
+    float tr_bot = currNoiseFunc(xHigh, yHigh, zLow);
+    float bl_bot = currNoiseFunc(xLow,  yLow,  zLow);
+    float br_bot = currNoiseFunc(xHigh, yLow,  zLow);
+    
+    // cant just divide all 
+    // interp1: need to interpolate between two of opp x with same y and same z loc (8 locations yield 4 locations)
+    // interp2: need to interpolate between two of x with opp y and same z loc (4 locations yield 2 locations)
+    // interp3: need to interpolate between two of x and y with opp z loc (2 locations yield 1 value)
+    
+    // interp value t set as .5 bc always directly halfway between the two locations
+    float t = 0.5;
+    
+    // interp1 //
+    float top_up = abs( cosInterpFunc(tl_up, tr_up, t) );
+    float back_up = abs( cosInterpFunc(bl_up, br_up, t) );
+    float top_bot = abs( cosInterpFunc(tl_bot, tr_bot, t) );
+    float back_bot = abs( cosInterpFunc(bl_bot, br_bot, t) );
+    
+    // interp2 //
+    float up = abs( cosInterpFunc(top_up, back_up, t) );
+    float bot = abs( cosInterpFunc(top_bot, back_bot, t) );
+    
+    // interp3 // TAKING ABS HERE NOW
+    float val = abs( cosInterpFunc(up, bot, t) );
```

Comment:
Absolute value here is also a bit odd. This will also result in not-smooth noise because of the jump from positive to negative